src/cpp/rtps/writer/StatefulWriter.cpp

```diff
@@ -142,13 +150,16 @@ void StatefulWriter::unsent_change_added_to_history(CacheChange_t* change)
                 if (m_separateSendingEnabled)
                 {
                     guids.at(0) = (*it)->m_att.guid;
-                    RTPSMessageGroup group(mp_RTPSParticipant, this, RTPSMessageGroup::WRITER, m_cdrmessages, 
-                        (*it)->m_att.endpoint.remoteLocatorList, guids);
-                    if (!group.add_data(*change, guids, (*it)->m_att.endpoint.remoteLocatorList, (*it)->m_att.expectsInlineQos))
+                    LocatorList_t remote_locators_shrinked{mp_RTPSParticipant->network_factory().ShrinkLocatorLists(
+                            {(*it)->m_att.endpoint.remoteLocatorList})};
+
```

Comment:
If I remeber correctly, remoteLocatorList is shrinked inside `matched_reader_add`, so there is no need to shrink it here

Reply:
Decided by the team to keep in this way.